scripts/get-prod-db.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+set -euxo
+
+# This script invokes an AWS Lambda function to retrieve a URL for downloading
+# a cleaned version of the production database and then restores
+# that data locally. By default the db name is "ynr-prod" but you can change the
+# local name by passing it as the first argument to the script.
+#
+# This script requires access to the YNR production AWS account
+#
+# Usage:
+#   ./script.sh [LOCAL_DB_NAME]
+#
+# Arguments:
+#   LOCAL_DB_NAME: Optional. Name of the local database to restore data to.
+#                  Defaults to 'ynr-prod' if not specified.
+
+# Configurable variables
+LAMBDA_FUNCTION_NAME="ynr-data-exporter"
+LOCAL_DB_NAME="${1:-ynr-prod}"
+
+# Check for required tools
+REQUIRED_TOOLS="aws dropdb createdb pg_restore wget"
+for tool in $REQUIRED_TOOLS; do
+  if ! command -v "$tool" >/dev/null 2>&1; then
+    echo "Error: $tool is required but not installed." >&2
+    exit 1
+  fi
+done
+
+# Create a temporary file and set up clean up on script exit
+TEMP_FILE=$(mktemp)
+trap 'rm -f "$TEMP_FILE"' EXIT
+
+# Invoke AWS Lambda and store the result in the temp file
+# The result is a presigned URL to the dump file on S3
+echo "Invoking Lambda to get DB URL. This might take a few minutes..."
+aws lambda invoke \
+  --function-name "$LAMBDA_FUNCTION_NAME" \
+  --cli-read-timeout=0 \
+  --no-cli-pager \
+  --output text \
+  --query 'Payload' \
+  "$TEMP_FILE"
+
+# Extract the URL from the response
+# This is because the response is quoted, so we just need to remove the quotation marks
+URL=$(sed 's/^"\(.*\)"$/\1/' "$TEMP_FILE")
```

Comment:
to make sure what we got back appears to be a URL. When I ran this, I got

```
URL={"errorMessage": "connection failed: connection to server at \"13.41.39.44\", port 5432 failed: FATAL:  database \"ynr-for-dev-export\" does not exist", "errorType": "OperationalError", "requestId": "9c15f87a-f4ca-4081-b3f5-ecd2a97aaf7c", "stackTrace": ["  File \"/app.py\", line 176, in lambda_handler\n    clean_database()\n", "  File \"/app.py\", line 74, in clean_database\n    conn = get_db_conn(db_name=TMP_DATABASE_NAME)\n", "  File \"/app.py\", line 32, in get_db_conn\n    conn = psycopg.connect(\n", "  File \"/usr/local/lib/python3.12/dist-packages/psycopg/connection.py\", line 119, in connect\n    raise last_ex.with_traceback(None)\n"]}
```

but the script just tried to plough on regardless

Reply:
I forgot to pull the commit out (I imagine we'll squash them all later anyway), but this is done here: https://github.com/DemocracyClub/yournextrepresentative/pull/2469/commits/51b35706346ae55da7dfb02e385935da7b1d293c#diff-388985bd50a1bb1aefbf516845f5019a2f244d5ba5cc1f10f0f6df6e98dddc1fR53